### ISSUE TITLE
module_defaults versus module property

### ISSUE THREAD

**Issue by ademprojects** [ansible/ansible#68992]
##### ISSUE TYPE
- Feature Idea
When providing module_defaults, this should be inherited up to the last element -> task.
But if you define more specific or even the property in a task, the property inside the task should win over module_defaults. This should be the expected goal.
Currently you get an error, that state is not defined, as module_defaults seems not to combine with loop_control.var.
Maybe this due to this reported bug ? https://github.com/ansible/ansible/issues/68991

##### COMPONENT NAME

module_defaults
##### ADDITIONAL INFORMATION

```yaml
    - package:
        name:  "{{ Ipackage.name |default(TaskName['defaults']['name']) }}"
        use:   "{{ Ipackage.use  |default(TaskName['defaults']['use'])  |default(omit) }}"
        state:   "{{ Ipackage.state |default(TaskName['defaults']['state']) }}"

      loop_control:
        loop_var: Ipackage
        label: "Module 'package': name='{{ Ipackage.name }}', {{Ipackage.state}}' "
      module_defaults:
        package:
          state: present
      << : *WDTaskDefaults #name and loop as YAML merge
```

**Comment by sivel** [comment]
I'm honestly not sure what your goal is, but as described in https://github.com/ansible/ansible/issues/68991, `module_defaults` does not impact `loop_control`.  That is simply not it's design.

Closing per https://github.com/ansible/ansible/issues/68991

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by ademprojects** [comment]
Hi Sivel,

the ticket is not a duplicate. The ticket describes inheritance and merging properties.

Providing **state** inside the task itself, **works fine**:
`
    - package:
        name:  "{{ Ipackage.name |default(TaskName['defaults']['name']) }}"
        use:   "{{ Ipackage.use  |default(TaskName['defaults']['use'])  |default(omit) }}"
        state:   "{{ Ipackage.state |default(TaskName['defaults']['state']) }}"
`
Providing **state** inside the task itself + **module_defaults.package.state** -> ERROR of duplicate property **state**:
`
    - package:
        name:  "{{ Ipackage.name |default(TaskName['defaults']['name']) }}"
        use:   "{{ Ipackage.use  |default(TaskName['defaults']['use'])  |default(omit) }}"
        state:   "{{ Ipackage.state |default(TaskName['defaults']['state']) }}"
      module_defaults:
        package:
          state: present
`
I think it makes more sense that state from the task should win over module_defaults. But I have to remove more specific properties in favour for module_defaults. That makes the useabilty and data deduplication less efficient. 

In other words, if you have big big big plays, and you set module_defaults inside of a play, than this one is inherited upto the task itself. But if one task has to set one property different, than you have to take this property out of module_defaults, and set to all tasks itself. means Tasks * property... a lot of unnecessary data, caused by the lack of merging with module_defaults.

**Comment by bcoca** [comment]
@ademprojects as stated in the other ticket, there is no variable or structure created that you can reference, you are making assumptions of how the internal objects  would be reflected as accessible structures, which is just not true, Ansible  does not expose (and never has) the internal playbook structures that way.
